packages/gatsby-theme-docs/src/components/link.spec.js

```diff
@@ -0,0 +1,134 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import {
+  createMemorySource,
+  createHistory,
+  LocationProvider,
+} from '@reach/router';
+import { useSiteData } from '../hooks/use-site-data';
+import Link from './link';
+
+jest.mock('../hooks/use-site-data');
+
+// for some types of tests you want a memory source
+const renderLink = (ui, initialPath) => {
+  const source = createMemorySource(initialPath);
+  const history = createHistory(source);
+  return render(<LocationProvider history={history}>{ui}</LocationProvider>);
+};
+
+const pathPrefix = '/prefix';
+const withPrefix = path => [pathPrefix, path].join('/').replace(/\/\/+/g, '/');
+
+describe('rendering', () => {
+  beforeEach(() => {
+    useSiteData.mockClear();
+    useSiteData.mockReturnValue({ pathPrefix });
+  });
+  const scenarios = [
+    {
+      title: 'image link',
+      props: { href: '/static/123' },
+      location: { pathname: '/page-1/' },
+      expected: { role: 'image-link' },
+    },
+    {
+      title: 'external link',
+      props: {
+        href: 'https://github.com/commercetools/commercetools-docs-kit',
+      },
+      location: { pathname: '/page-1/' },
+      expected: { role: 'external-link' },
```

Comment:
I would prefer explicitly testing the href output in all test cases and explicitly for prod vs. dev.  
It's also for making sure that all links actually write a href so that  opening in a new tab always works. 

And I tend to not trust test implementations that do too much magic in own code to generate assertions for cases that are not actually "linear" in the sense that all cases are handled the same.

Reply:
The prod links will be tested as e2e tests (which is what is currently missing in this PR).